hudi-client/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -159,24 +161,32 @@ private void rollBackIndex() {
     LOG.info("Index rolled back for commits " + instantToRollback);
   }
 
-  public List<HoodieRollbackStat> doRollbackAndGetStats() {
-    final String instantTimeToRollback = instantToRollback.getTimestamp();
-    final boolean isPendingCompaction = Objects.equals(HoodieTimeline.COMPACTION_ACTION, instantToRollback.getAction())
-        && !instantToRollback.isCompleted();
-    validateSavepointRollbacks();
-    if (!isPendingCompaction) {
-      validateRollbackCommitSequence();
-    }
-
-    try {
-      List<HoodieRollbackStat> stats = executeRollback();
-      LOG.info("Rolled back inflight instant " + instantTimeToRollback);
+  public List<HoodieRollbackStat> mayBeRollbackAndGetStats(boolean doDelete) {
```

Comment:
@vinothchandar : I have added a flag here to say where delete has to be done or just stats need to be collected. Since I don't want to duplicate code, tried my best to re-use. If you can think of any other ways, lmk.